File: src/main/java/at/searles/regexp/NonGreedy.java
package at.searles.regexp;

/**
 * Strictly speaking not a regex, but a
 * good and efficient replacement for non-greedy.
 */
class NonGreedy implements Regexp {

    private final Regexp t;

    NonGreedy(Regexp t) {
        this.t = t;
    }

    @Override
    public <A> A accept(Visitor<A> visitor) {
        return visitor.visitNonGreedy(t);
    }

    @Override
    public String toString() {
        return String.format("nonGreedy(%s)", t);
    }
}

File: src/main/java/at/searles/regexp/NonGreedy.kt
package at.searles.regexp

/**
 * Strictly speaking not a regex, but a
 * good and efficient replacement for non-greedy.
 */
internal class NonGreedy(private val t: Regexp) : Regexp {
    override fun <A> accept(visitor: Visitor<A>): A {
        return visitor.visitNonGreedy(t)
    }

    override fun toString(): String {
        return String.format("nonGreedy(%s)", t)
    }

}